opengrok-web/src/main/java/org/opengrok/web/api/v1/filter/IncomingFilter.java

```diff
@@ -86,14 +97,38 @@ public void filter(final ContainerRequestContext context) {
         if (request == null) { // happens in tests
             return;
         }
+
         String path = context.getUriInfo().getPath();
+
+        if (request.isSecure()) {
+            String authHeaderValue;
+            if ((authHeaderValue = request.getHeader(HttpHeaders.AUTHORIZATION)) != null &&
+                    authHeaderValue.startsWith(BEARER)) {
+                String tokenValue = authHeaderValue.substring(BEARER.length());
+                if (RuntimeEnvironment.getInstance().getAuthenticationTokens().contains(tokenValue)) {
```

Comment:
could benefit from ConfiguratioChangeListener (https://github.com/oracle/opengrok/pull/3270) so you could cache the tokens as it requires a lock on each secure request which contains the header

Reply:
good idea. Will wait for the PR to land.

Reply:
@ahornace was so kind to allow me to take the `ConfiguratioChangedListener` from his PR.